src/lib.rs

```diff
@@ -228,15 +128,47 @@ impl CrashReporter {
         );
         data.insert("version".to_string(), json!(report.version));
 
-        if let Some(threshold_value) = &self.threshold {
-            data.insert("threshold".to_string(), json!(threshold_value));
+        if data
+            .get("errorGroups")
```

Comment:
I'm sorry, I am rusty on my Rust. But can't you use pattern matching to check for optionals? 

Something like
```
match data.get("errorGroups") {
    None => {
        // select the empty template
    }
}

Reply:
Then you can can get rid of all the unwrapping, arraying, unwrapping.

Reply:
It appers to me that this is not possible.
Reason: 
```match data.get("errorGroups") {
    None => {```

Is not none if AppCenter has no crashes for a version. No crashes are modeled as an empty array.

Reply:
I'm guessing you can also do pattern matching with an empty array, but it's only a guess.